lib/document.js

```diff
@@ -2505,6 +2505,14 @@ function _getPathsToValidate(doc) {
       continue;
     }
 
+    if (_pathType.$isMongooseDocumentArray) {
+      for (const p of paths) {
+        if (p === null || p.startsWith(_pathType.path + '.')) {
```

Comment:
Is strict equality in `p === null` intended, or did you mean `p == null`?

Reply:
In ine 2487 we also have strill null check